cadc-auth-restlet/src/test/java/ca/nrc/cadc/auth/restlet/RestletPrincipalExtractorTest.java

```diff
@@ -63,6 +64,54 @@
 
     private RestletPrincipalExtractor testSubject;
     private final Request mockRequest = createMock(Request.class);
+    
+    @Test
+    public void testCaseInsensitiveDelegationToken() throws Exception {
+    	// create an invalid token and test that RestletPrincipalExtractor finds it even when case of
+    	// attribute is changed
+        setTestSubject(new RestletPrincipalExtractor() {
+            @Override
+            public Request getRequest() {
+                return getMockRequest();
+            }
+
+            @Override
+            protected String getAuthenticatedUsername() {
+                return null;
+            }
+        });
+ 
+        final ConcurrentMap<String, Object> attributes
+                = new ConcurrentHashMap<String, java.lang.Object>();
+        Form form = new Form(AuthenticationUtil.AUTH_HEADER + "=foo");
+        attributes.put("org.restlet.http.headers", form);
+        expect(getMockRequest().getAttributes()).andReturn(attributes).atLeastOnce();
+
+        replay(getMockRequest());
+        try {
+        		DelegationToken dt = getTestSubject().getDelegationToken();
+        		assertTrue(false);
+        } catch (NotAuthenticatedException e)
+        {
+        		assertEquals("Unexpected exception", "invalid delegation token. null", e.getMessage());
+        }
+        
+        // repeat test with lowercase attributes
+        reset(getMockRequest());
+        attributes.clear();
+        form = new Form(AuthenticationUtil.AUTH_HEADER.toLowerCase() + "=foo");
+        attributes.put("org.restlet.http.headers", form);
+        expect(getMockRequest().getAttributes()).andReturn(attributes).atLeastOnce();
+        
+		replay(getMockRequest());
+		try {
+				DelegationToken dt = getTestSubject().getDelegationToken();
+				assertTrue(false);
+		} catch (NotAuthenticatedException e)
+		{
+				assertEquals("Unexpected exception", "invalid delegation token. null", e.getMessage());
```

Comment:
The message needs to say "expected exception" so it is clear why it is being caught and dealt with in this way.